packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts

```diff
@@ -235,66 +288,100 @@ export class SummarizerNode implements IRootSummarizerNode {
 		readAndParseBlob: ReadAndParseBlob,
 		correlatedSummaryLogger: ITelemetryLogger,
 	): Promise<RefreshSummaryResult> {
-		this.defaultLogger.sendTelemetryEvent({
-			eventName: "refreshLatestSummary_start",
-			proposalHandle,
-			referenceSequenceNumber: this.referenceSequenceNumber,
-			summaryRefSeq,
-		});
+		/**
+		 * Refresh latest summary should never happen while a summary is in progress. If it does, log an error
+		 * so that we can gather data around it and take appropriate action.
+		 */
+		if (this.isTrackingInProgress()) {
+			this.logger.sendErrorEvent({
+				eventName: "UnexpectedRefreshDuringSummarize",
+				proposalHandle,
+				summaryRefSeq,
+				inProgressSummaryRefSeq: this.wipReferenceSequenceNumber,
+			});
+		}
 
-		if (proposalHandle !== undefined) {
-			const maybeSummaryNode = this.pendingSummaries.get(proposalHandle);
+		let result: RefreshSummaryResult | undefined;
+		try {
```

Comment:
The change here is the addition of try ... finally block to add the "refreshLatestSummary_end" log at the end. There is no change in the logic.